sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs

```diff
@@ -252,8 +256,25 @@ public virtual async Task<Response> GetLedgerEntryAsync(string transactionId, st
             scope.Start();
             try
             {
-                using HttpMessage message = CreateGetLedgerEntryRequest(transactionId, collectionId, context);
-                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+                try
+                {
+                    using HttpMessage primaryMessage = CreateGetLedgerEntryRequest(_ledgerEndpoint, transactionId, collectionId, context);
+                    return await _pipeline.ProcessMessageAsync(primaryMessage, context).ConfigureAwait(false);
+                }
+                catch (Exception)
+                {
+                    Response failoverCurrent = await _failoverService.ExecuteOnFailoversAsync(
+                        _ledgerEndpoint,
+                        async (endpoint) =>
+                        {
+                            using HttpMessage message = CreateGetCurrentLedgerEntryRequest(endpoint, collectionId, context);
+                            return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+                        },
+                        nameof(GetCurrentLedgerEntryAsync),
+                        collectionId,
+                        context?.CancellationToken ?? default).ConfigureAwait(false);
```

Comment:
To make sure failover works as expected the timeout would need to be extended, for instance if the client is getting an entry the timer starts. The requests will be retried to primary according to the client options and default pipeline options, i.e. this will eat into timeout quite a bit. Then we say the failover needs to happen with the remaining time which not only needs to fulfill the original intent of getting an entry but also fetch the list of failover ledgers (needs more work).

What I am saying is that we either automatically extend the timeout here, or increase it in the default client options for all ledgers that use failover endpoints or somehow tell the customers to make sure it is increased in their client options to work as expected.